pkg/health/probe/responder/responder.go

```diff
@@ -17,29 +18,62 @@ var defaultTimeout = 30 * time.Second
 
 // Server wraps a minimal http server for the /hello endpoint
 type Server struct {
-	httpServer http.Server
+	httpServer []*http.Server
 }
 
 // NewServer creates a new server listening on the given port
-func NewServer(port int) *Server {
-	return &Server{
-		http.Server{
-			Addr:    fmt.Sprintf(":%d", port),
+func NewServer(address []string, port int) *Server {
+	if len(address) == 0 {
+		address = []string{""}
+	}
+
+	server := &Server{}
+	for _, ip := range address {
+		if ip != "" {
+			ipBytes, _ := netip.ParseAddr(ip)
+			if ipBytes.Is6() {
+				// if ipv6 address, then listen address should be in format of [ipv6]:port
+				ip = "[" + ip + "]"
+			}
+		}
+
+		hs := http.Server{
+			Addr:    fmt.Sprintf("%s:%d", ip, port),
 			Handler: http.HandlerFunc(serverRequests),
-		},
+		}
+		server.httpServer = append(server.httpServer, &hs)
 	}
+
+	return server
 }
 
 // Serve http requests until shut down
 func (s *Server) Serve() error {
-	return s.httpServer.ListenAndServe()
+	errors := make(chan error)
+	for _, hs := range s.httpServer {
+		tmpHttpServer := hs
+		go func() {
+			errors <- tmpHttpServer.ListenAndServe()
+		}()
+	}
+
+	// Block for the first error, then return.
+	err := <-errors
+	return err
 }
 
 // Shutdown server gracefully
 func (s *Server) Shutdown() error {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
 	defer cancel()
-	return s.httpServer.Shutdown(ctx)
+	var err error
+	for _, hs := range s.httpServer {
+		if tmpError := hs.Shutdown(ctx); tmpError != nil {
+			err = tmpError
+		}
+	}
+
+	return err
```

Comment:
I suggest to join all the errors like this:

```go
	errs := make([]error, 0, len(s.httpServer))
	for _, hs := range s.httpServer {
		errs = append(errs, hs.Shutdown(ctx))
	}
	return errors.Join(errs...)
```
[errors,Join](https://pkg.go.dev/errors#Join) will discard any nil error, if any.